bundles/binding/org.openhab.binding.fatekplc/src/main/java/org/openhab/binding/fatekplc/internal/FatekPLCBinding.java

```diff
@@ -0,0 +1,278 @@
+				if (slave == null) {
+					slave = new FatekPLCSlave(keyItems[0], eventPublisher, itemRegistry);
+					slaves.put(slave.getName(), slave);
+				}
+
+				slave.configure(keyItems[1], properties.get(key));
+			}
+
+		} catch (ConfigurationException e) {
+			logger.error("", e);
+			throw e;
+		}
+
+		setProperlyConfigured(true);
+		logger.info("Configuration has been updated, refresh={}", refreshInterval);
+	}
+
+	/**
+	 * Parse long from property with default.
+	 *
+	 * @param properties
+	 *            properties map
+	 * @param key
+	 *            given key
+	 * @param defaultVale
+	 * @return long value for given key
+	 */
+	private long getLongProperty(Dictionary<String, ?> properties, String key, long defaultVale) {
+
+		String value = (String) properties.get(key);
+		long ret;
+		try {
+			ret = Long.parseLong(value);
+		} catch (NumberFormatException e) {
+			ret = defaultVale;
+			logger.warn("Property {} parse exception {}", key, e.getMessage());
+		}
+
+		return ret;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+
+		Map<String, List<FatekPLCItem>> toUpdate = new HashMap<>();
+
+		// collect items by slave PLC
+		for (FatekPLCBindingProvider provider : providers) {
+			for (String itemName : provider.getItemNames()) {
+
+				FatekPLCItem item = provider.geFatektItem(itemName);
+
+				if (item.isToRefresh()) {
+
+					List<FatekPLCItem> itemsList = toUpdate.get(item.getSlaveName());
+					if (itemsList == null) {
+						itemsList = new ArrayList<>();
+						toUpdate.put(item.getSlaveName(), itemsList);
+					}
+
+					itemsList.add(item);
+				}
+			}
+		}
+
+		logger.debug("toUpdate={}", toUpdate);
+
+		// process list to update
+		for (String slaveName : toUpdate.keySet()) {
+
+			FatekPLCSlave slave = slaves.get(slaveName);
+			List<FatekPLCItem> items = toUpdate.get(slaveName);
+
+			if (slave != null) {
+				try {
+					slave.updateItems(items);
+				} catch (Exception e) {
+					logger.error("update items error", e);
+				}
+			} else {
+				logger.warn("Unknown slave: {} to process update: {}", slaveName, items);
+			}
+		}
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void internalReceiveCommand(String itemName, Command command) {
+
+		logger.debug("internalReceiveCommand({},{}) is called!", itemName, command);
+
+		for (FatekPLCBindingProvider provider : providers) {
+			FatekPLCItem item = provider.geFatektItem(itemName);
+			if (item != null) {
+				FatekPLCSlave slave = slaves.get(item.getSlaveName());
+				if (slave != null) {
+					try {
+						slave.command(item, command);
+					} catch (Exception e) {
+						logger.error(String.format("internalReceiveCommand(%s, %s)", itemName, command), e);
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void internalReceiveUpdate(String itemName, State newState) {
+		logger.debug("internalReceiveUpdate({},{}) is called!", itemName, newState);
+	}
+}
```

Comment:
Since the binding does not handle item state updates, this override can be removed.